masterslave.q

```diff
@@ -0,0 +1,109 @@
+//Master slave algorithm
+//Each process initially assumes it's the master process
+//Master process is determined by oldest process start time
+//Connect to other slave processes and retrieve details to determine which process is master
+
+//run lines
+//q torq.q -load masterslave.q -proctype lb -procname lb1 -p 6100 -debug -parentproctype wdb
+//q torq.q -load masterslave.q -proctype lb -procname lb2 -p 6101 -debug -parentproctype wdb
+
+//test cases
+//one process starts up, it becomes master
+//two processes, only one is master
+//two processes, master dies, slave takes over
+//two processes, one is made master, new process added, original master(if it's starttime
+// is still oldest)
+//two processes, one is made master, an older process is connected, recently connected process
+// should then become master
+//3 processes, one master, master dies, one of the remaining two takes over
+
+
+\d .masterslave
+init:{
+ // the list of processes to connect to
+ .servers.CONNECTIONS:distinct .servers.CONNECTIONS,.proc.proctype;
+
+ // custom function, this is invoked when a new outbound connection is created
+ // to be customised to invoke negotiation of processes
+ .servers.connectcustom:{[f;x] .masterslave.checkifmaster[];
```

Comment:
I think this will work, but there is still too much work going on here (too much IPC)
Basically .servers.connectcustom will get invoked when a new connection is made, and it will be passed in (as the x argument in this case) the new connections. So, you only need this process to negotiate with those new processes, not all the old ones too. In this version, if there are 10 connections, and a new one arrives, then the way you have it every process will make 11 outbound requests, whereas really they only need to be making one

Reply:
Is `f` the function to be called on the new connection handle`x`?

And instead of running .ms.checkifmaster when a new connection is made,
I should call .ms.addmember with the new connection procname?

Reply:
> Is f the function to be called on the new connection handlex?
Yes. f is the pre-existing definition of .servers.connectcustom (which may be nothing)
> I should call .ms.addmember with the new connection procname
I think so